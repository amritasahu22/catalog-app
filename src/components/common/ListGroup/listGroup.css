.list-group-ul {
	overflow: auto;
	max-height: 300px;
}
