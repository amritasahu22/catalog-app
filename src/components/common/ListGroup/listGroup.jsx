import React from "react";
import PropTypes from "prop-types";
import "./listGroup.css";

const ListGroup = (props) => {
	const {
		items,
		keyProperty,
		textProperty,
		selectedItem,
		onItemSelect,
	} = props;

	const styles = {
		li: {
			fontSize: "0.9rem",
			padding: "0.1rem",
			fontFamily: "san-serif",
		},
	};

	return (
		<ul className="list-group clickable list-group-ul">
			{items.map((item) => (
				<li
					key={item[keyProperty]}
					className={
						item === selectedItem
							? "list-group-item list-group-item-action list-group-item-light border-white active"
							: "list-group-item list-group-item-action list-group-item-light border-white"
					}
					style={styles.li}
					onClick={() => onItemSelect(item)}
				>
					{item[textProperty]}
				</li>
			))}
		</ul>
	);
};

ListGroup.defaultProps = {
	keyProperty: "id",
	textProperty: "name",
};

ListGroup.propTypes = {
	items: PropTypes.array.isRequired,
	onItemSelect: PropTypes.func.isRequired,
};

export default ListGroup;
