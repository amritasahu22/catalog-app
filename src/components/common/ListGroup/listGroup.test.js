import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ListGroup from "./listGroup";

describe("ListGroup", () => {
	const brands = [
		{
			id: 1,
			name: "17 Sundays",
			url_key: "17-sundays",
		},
		{
			id: 2,
			name: "2XU",
			url_key: "2xu",
		},
	];

	test("should render properly with props", () => {
		const component = shallow(
			<ListGroup
				items={brands}
				selectedItem={brands[0]}
				onItemSelect={() => {}}
			/>
		);
		expect(toJson(component)).toMatchSnapshot();
	});
});
