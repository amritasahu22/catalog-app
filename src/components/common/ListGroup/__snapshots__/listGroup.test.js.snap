// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup should render properly with props 1`] = `
<ul
  className="list-group clickable list-group-ul"
>
  <li
    className="list-group-item list-group-item-action list-group-item-light border-white active"
    key="1"
    onClick={[Function]}
    style={
      Object {
        "fontFamily": "san-serif",
        "fontSize": "0.9rem",
        "padding": "0.1rem",
      }
    }
  >
    17 Sundays
  </li>
  <li
    className="list-group-item list-group-item-action list-group-item-light border-white"
    key="2"
    onClick={[Function]}
    style={
      Object {
        "fontFamily": "san-serif",
        "fontSize": "0.9rem",
        "padding": "0.1rem",
      }
    }
  >
    2XU
  </li>
</ul>
`;
